parachain/pallets/ethereum-beacon-client/src/merkleization.rs

```diff
@@ -4,7 +4,7 @@ use crate::{config, ssz::*};
 use byte_slice_cast::AsByteSlice;
 use frame_support::{traits::Get, BoundedVec};
 use snowbridge_beacon_primitives::{
-	BeaconHeader, ExecutionPayload, ForkData, SigningData, SyncAggregate, SyncCommittee,
+	BeaconHeader, ExecutionPayloadHeader, ForkData, SigningData, SyncAggregate, SyncCommittee,
```

Comment:
Nit: This should probably just be `ExecutionHeader`. It is either `ExecutionPayload` (containing all the transactions) or `ExecutionHeader` containing the transactions root.

Reply:
Yes, the previous name `ExectionPayLoad` is a little confusing which actually contains `transactions_root` [here](https://github.com/Snowfork/snowbridge/blob/c3bd614dffad1dae5996b8b67e6cfd3023669111/parachain/primitives/beacon/src/lib.rs#L410) so I rename it as `ExectionPayLoadHeader`. I would assume after #783 we do not need `ExecutionPayload`  (containing all the transactions) any more but correct me if I'am wrong.

And about `ExecutionHeader`, AFAIK not all fields in `ExectionPayLoadHeader` are required for proof verification. e.g only `receipts_root` is required [here](https://github.com/Snowfork/snowbridge/blob/f5dc322860015c6e711c82e53a273d9b9d6764c2/parachain/pallets/ethereum-beacon-client/src/lib.rs#L1129) for execution layer message, so I just simplify [ExecutionHeader](https://github.com/Snowfork/snowbridge/blob/f5dc322860015c6e711c82e53a273d9b9d6764c2/parachain/primitives/beacon/src/lib.rs#L291) as a subset of `ExectionPayLoadHeader` and make it an version insensitive/agnostic data structure.

Reply:
> And about ExecutionHeader, AFAIK not all fields in ExectionPayLoadHeader are required for proof verification. e.g only receipts_root is required [here](https://github.com/Snowfork/snowbridge/blob/f5dc322860015c6e711c82e53a273d9b9d6764c2/parachain/pallets/ethereum-beacon-client/src/lib.rs#L1129) for execution layer message, so in this PR I just simplify [ExecutionHeader](https://github.com/Snowfork/snowbridge/blob/f5dc322860015c6e711c82e53a273d9b9d6764c2/parachain/primitives/beacon/src/lib.rs#L291) as a subset of ExectionPayLoadHeader and make it an version insensitive/agnostic data structure.

I think this is a good optimization to have, as it will cut down on storage costs in the BridgeHub parachain. So I support it.

Reply:
Agreed, thanks @yrong!